src/algebra/hom/iterate.lean

```diff
@@ -73,6 +73,14 @@ commute.iterate_left (λ x, f.map_zpow x m) n a
 lemma coe_pow {M} [comm_monoid M] (f : monoid.End M) (n : ℕ) : ⇑(f^n) = (f^[n]) :=
 hom_coe_pow _ rfl (λ f g, rfl) _ _
 
+@[simp, to_additive]
+lemma iterate_zpow (n : ℤ) (j : ℕ) : ((λ (x : G), x^n)^[j]) = λ x, x^(n^j) :=
```

Comment:
I think this is in the wrong file, as it's not about *_`hom`s at all. Do we have the `pow` version of this lemma somewhere else?

Reply:
Ah; this is the right file, but the lemma belongs next to [`mul_right_iterate`](https://leanprover-community.github.io/mathlib_docs/algebra/hom/iterate.html#mul_right_iterate), probably as `pow_right_iterate`. The current `monoid_hom.iterate_pow` is definitely wrong.

Reply:
Thanks, I've moved the lemma lower down, very close to `mul_right_iterate`. I called it `pow_iterate` though rather than `pow_right_iterate` which I think fits better with our conventions (correct me if I'm wrong!). I also added the version I need called `zpow_iterate`.